### PR TITLE
feat(app): DeckCalibration Section

### LINKED ISSUE #8759: Calibration - Deck Calibration
# Overview
The second section on the robot settings page's calibration tab is used to explain deck calibration to the user. 
We should show the user when the last deck calibration was performed, and allow the user to click a "Recalibrate deck" button that will kick off a new deck calibration workflow. 

# Design
This artboard shows Deck calibration complete, the Deck calibration needed error, and the Deck calibration needed warning: 
https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A90420

# Acceptance Criteria
- [ ] The "Recalibrate deck" button should kick off the deck calibration workflow

Most recent deck calibration: 
- [ ] If deck calibration has been performed, the date and time of the most recent deck calibration should appear correctly

- [ ] If deck calibration has not been performed yet, "Not calibrated yet" should be shown where the date and time normally appear

Errors and warnings: 
- [ ] If deck calibration hasn't been performed yet, a red error banner should display "Calibration needed" and the button should say "Calibrate deck" (instead of "Recalibrate deck" 

- [ ] If calibration health check returns any problems, a yellow warning banner should display "Recalibration recommended"

### PR DESCRIPTION
**Pull request by koji**
# Overview
Break down RobotSettings Calibration Tab into a small PR
This will fix  #8759 #8796

Screen Shot 2022-05-09 at 4 12 48 PM

Screen Shot 2022-05-09 at 5 42 25 PM

# Changelog
- Add Deck Calibration section to RobotSettings Calibration tab

# Review requests
- When a robot hasn't done deck calibration, the button is `Calibrate deck`
- After deck calibration the calibration button is `Recalibrate deck`
- When clicking the deck calibration button, the app shows a modal which is the same as 5.0.2

# Risk assessment

### CODE REVIEW

**Comment by koji** [comment]
- [ ] Add test cases
  - [ ]  yet deck cal `calibrate deck`
  - [ ]  yet deck cal shows banner
  - [ ]  already done deck cal `recalibrate deck`
- [ ]  click `calibrate deck` call a function
- [ ]  click  `recalibrate deck` call a function
- [ ]  click `calibrate now` call a function

**Comment by koji** [comment]
fixing the issue
- [x] fix component
- [x] fix test cases

File: app/src/organisms/Devices/RobotSettings/RobotSettingsCalibration.tsx [hunk 1]
```diff
@@ -10,42 +11,154 @@ import {
...
+      : false
+
+  const deckCalibrationSession: DeckCalibrationSession | null = useSelector(
+    (state: State) => {
+      return getDeckCalibrationSession(state, robotName)
+    }
+  )
+
+  const {
+    // showConfirmation: showConfirmStart,
```
Comment (b-cooper) [inline app/src/organisms/Devices/RobotSettings/RobotSettingsCalibration.tsx]: is there a reason why these are commented out?

**Review by b-cooper — APPROVED** [review]
Changes look good ! 🚋

**Comment by koji** [comment]
- [x] fix conflicts
- [x] fix feedback 
- [x] fix check-js issue

File: app/src/organisms/Devices/RobotSettings/RobotSettingsCalibration.tsx (hunk 1 above)
Comment (koji) [reply app/src/organisms/Devices/RobotSettings/RobotSettingsCalibration.tsx]: Seems that I forgot to put back a new modal I added.
imported it to RobotSettingsCalibration and pass them to the modal.

**Pull request merged**